packages/material-ui-lab/src/PaginationItem/PaginationItem.js

```diff
@@ -0,0 +1,393 @@
+      // Reset on touch devices, it doesn't add specificity
+      '@media (hover: none)': {
+        backgroundColor: 'transparent',
+      },
+    },
+    '&$selected': {
+      color: theme.palette.secondary.main,
+      border: `1px solid ${fade(theme.palette.secondary.main, 0.5)}`,
+      backgroundColor: fade(theme.palette.secondary.main, 0.15),
+      '&:hover, &:focus': {
+        backgroundColor: fade(theme.palette.secondary.dark, 0.17),
+        // Reset on touch devices, it doesn't add specificity
+        '@media (hover: none)': {
+          backgroundColor: 'transparent',
+        },
+      },
+    },
+  },
+  /* Styles applied to the button element if `rounded="true"`. */
+  rounded: {
+    borderRadius: theme.shape.borderRadius,
+  },
+  /* Styles applied to the ellipsis element. */
+  ellipsis: {
+    fontSize: theme.typography.pxToRem(14),
+    textAlign: 'center',
+    width: 38,
+    '&$disabled': {
+      color: fade(theme.palette.text.primary, 0.5),
+    },
+    '&$sizeSmall': {
+      fontSize: theme.typography.pxToRem(13),
+      width: 28,
+    },
+    '&$sizeLarge': {
+      fontSize: theme.typography.pxToRem(15),
+      width: 48,
+    },
+  },
+  /* Styles applied to the icon element. */
+  icon: {
+    fontSize: theme.typography.pxToRem(20),
+    '&$sizeSmall': {
+      fontSize: theme.typography.pxToRem(18),
+      width: 28,
+    },
+    '&$sizeLarge': {
+      fontSize: theme.typography.pxToRem(22),
+      width: 48,
+    },
+  },
+  /* Pseudo-class applied to the root element if `size="small"`. */
+  sizeSmall: {},
+  /* Pseudo-class applied to the root element if `size="large"`. */
+  sizeLarge: {},
+  /* Pseudo-class applied to the root element if `disabled={true}`. */
+  disabled: {},
+  /* Pseudo-class applied to the root element if `selected={true}`. */
+  selected: {},
+});
+
+function ariaLabel(type, page, selected) {
+  if (type === 'page') {
+    return `${selected ? '' : 'go to '}page ${page}`;
+  }
+  return `Go to ${type} page`;
+}
+
+const PaginationItem = React.forwardRef(function PaginationItem(props, ref) {
+  const {
+    classes,
+    className,
+    color = 'standard',
+    component,
+    disabled = false,
+    getAriaLabel,
+    page,
+    onClick: handleClick,
+    selected,
+    shape = 'round',
+    size = 'medium',
+    type = 'page',
+    variant,
+    ...other
+  } = props;
+
+  return type === 'start-ellipsis' || type === 'end-ellipsis' ? (
+    <div
+      ref={ref}
+      className={clsx(classes.ellipsis, {
+        [classes.disabled]: disabled,
+        [classes[`size${capitalize(size)}`]]: size !== 'medium',
+      })}
+    >
+      …
+    </div>
+  ) : (
+    <ButtonBase
+      ref={ref}
+      component={component}
+      disabled={disabled}
+      aria-label={
+        getAriaLabel ? getAriaLabel(type, page, selected) : ariaLabel(type, page, selected)
+      }
+      aria-current={selected ? 'page' : undefined}
```

Comment:
I'm a bit uncertain if this is intended usage. I only ever saw this used in anchors. I'm not sure how this behaves on buttons. Unless we're sure about the impact it has I would rather not use it (1st rule of aria). In the end it's trivial to supply this as a user since they already know if a PaginationItem is current i.e. `selected={true}`.

Reply:
I'm going to have to start keeping a better record of my research, as this isn't the first time I've done or something based on a review of the literature, but haven't recorded the findings that led to the decision in the first place!

I'll take another look, and if I can't find suitable justification, I'll remove it (I agree with 1st rule).

Reply:
So... The spec refers to an element, without defining any specific type of element:
> **aria-current** (state)
> Indicates the element that represents the current item within a container or set of related elements.

*https://www.w3.org/TR/wai-aria-1.1/#aria-current*

Here's an example of using it on `td`:
http://design-patterns.tink.uk/aria-current/

And the NVDA team seemed to view browsers not supporting it on non-link elements as a bug: https://github.com/nvaccess/nvda/issues/6358#issuecomment-302076134

So, I think we're good, but happy to take feedback to the contrary.

Reply:
My main concern is the kind of value we pass here. As I understood it `aria-current="page"` is only appropriate for anchors while we have a set of buttons. It seems like `aria-current="true"` would be better here?

Reply:
> As I understood it aria-current="page" is only appropriate for anchors 

Indeed, and from the w3 page: "For example: A page token used to indicate a link within a set of pagination links"

But that's an example, and the preceding text again refers generically to "an element": 

> The aria-current attribute is used when an element within a set of related elements is visually styled to indicate it is the current item in the set.

The "page" value for aria-current is an additional affordance that indicates what type of item is "current", not what type of element it's attached to.

That said, it should already be abundantly clear from other ARIA attributes that this is a pagination control, and from the button ARIA lables what the buttons represent, so changing the value to `true` should be no great loss.

Reply:
> The "page" value for aria-current is an additional affordance that indicates what type of item is "current", not what type of element it's attached to.

This is not how I read the spec:
> A page token used to indicate a link within a set of pagination links, where the link is visually styled to represent the currently-displayed page.

If the user does not use the component as a link (which is the current default) then `aria-current="page"` would be invalid, no?

Reply:
> This is not how I read the spec

It's the same sentence I quoted, but without the preceding words "For example". If the attribute and value can be applied to any element, and that's just an example, I don't see how the spec precludes using it for other elements.

Anyway, we're just arguing semantics at this point, since I'd already accepted that regardless of what the spec does or doesn't say, it probably isn't necessary to specify the particular `aria-current` type, and had already changed it to the more generic (but still perfectly valid) `true' that you had previously suggested.

Reply:
> don't see how the spec precludes using it for other elements.

It explicitly says that the **value** `page` should be used for links. It doesn't explicitly say to not use it for other elements but that interpretation makes the spec meaningless since it would require to allow "X" **and** disallow "not X" for every definition. In other words: with your interpretation we could use `date` as well because the spec doesn't say that we can't.

I'm not arguing about the **attribute** usage. That is clearly allowed for arbitrary sets of elements. What is not valid usage is using the **value** `page` in a way that is not described in the spec.